scripts/production/batch_lastz.pl

```diff
@@ -83,68 +83,40 @@
-    	} else {
-		    $ref_chunk_count = get_ref_chunk_count($mlss_gdbs->[1]);
-	    	$non_ref_chunk_count = get_non_ref_chunk_count($mlss_gdbs->[0]);
-    	}
-        $filter_dups_job_count = get_ref_chunk_count($mlss_gdbs->[0]) + get_ref_chunk_count($mlss_gdbs->[1]);
-
-        # my ( $dnaf_count_1, $dnaf_count_2 ) = (chains_dnafrag_count($mlss_gdbs->[0]), chains_dnafrag_count($mlss_gdbs->[1]));
-        # my ( $gdb_name_1, $gdb_name_2 ) = ( $mlss_gdbs->[0]->name, $mlss_gdbs->[1]->name );
-        # print "dnaf_count $gdb_name_1: $dnaf_count_1; dnaf_count $gdb_name_2: $dnaf_count_2\n";
-        # my $chains_job_count = $dnaf_count_1 * $dnaf_count_2;
+        # For polyploid PWAs of the same genus, this makes an overestimate as
+        # they will not share all the components
+        $filter_dups_job_count = $ref_chunk_count + get_ref_chunk_count($mlss_gdbs[1]);
+    }
+    # Only non-polyploid self-alignments do not produce chain or net jobs
+    if (($mlss_gdbs[0]->name ne $mlss_gdbs[1]->name) || $mlss_gdbs[0]->is_polyploid) {
         my $chains_job_count = chains_job_count( $mlss );
-        # my $chains_job_count = ceil($lastz_job_count/3); # generally, this is ~0.3333 of LastZ jobs
-        $mlss_job_count{$mlss->dbID}->{aln_chains} = $chains_job_count;
-        #$total_job_count{aln_chains} += $chains_job_count;
-        $mlss_job_count{$mlss->dbID}->{aln_nets} = ceil($chains_job_count/2);
-        #$total_job_count{aln_nets} += ceil($chains_job_count/2); # alignment_nets = ~ half chain jobs
-
-        $mlss_job_count{$mlss->dbID}->{self_aln} = 0;
+        $mlss_job_count{$mlss->dbID}->{analysis}->{aln_chains} = $chains_job_count;
+        $mlss_job_count{$mlss->dbID}->{analysis}->{aln_nets} = ceil($chains_job_count/2);
     }
-
 	my $lastz_job_count = ($ref_chunk_count * $non_ref_chunk_count);
-	$mlss_job_count{$mlss->dbID}->{lastz} = $lastz_job_count;
-	# $total_job_count{lastz} += $lastz_job_count;
+    $mlss_job_count{$mlss->dbID}->{analysis}->{lastz} = $lastz_job_count;
+    $mlss_job_count{$mlss->dbID}->{analysis}->{filter_dups} = $filter_dups_job_count;
+    # dump_large_nib_for_chains and coding_exon_stats are usually around the same value
+    $mlss_job_count{$mlss->dbID}->{analysis}->{dump_nibs} = $filter_dups_job_count;
+    $mlss_job_count{$mlss->dbID}->{analysis}->{exon_stats} = $filter_dups_job_count;
 
-	$mlss_job_count{$mlss->dbID}->{filter_dups} = $filter_dups_job_count;
-	# $total_job_count{filter_dups} += $filter_dups_job_count;
-	# dump_large_nib_for_chains and coding_exon_stats are usually around the same value
-	$mlss_job_count{$mlss->dbID}->{dump_nibs} = $filter_dups_job_count;
-	# $total_job_count{dump_nibs} += $filter_dups_job_count;
-	$mlss_job_count{$mlss->dbID}->{exon_stats} = $filter_dups_job_count;
-	# $total_job_count{exon_stats} += $filter_dups_job_count;
+    $mlss_job_count{$k}->{all} = sum(values %{ $mlss_job_count{$mlss->dbID}->{analysis} });
```

Comment:
Does this script compile? I don't think `$k` is defined anywhere?

Reply:
Weirdly enough, it passes `perlcritic`. Well spotted that bug! (damn copy-paste...)

Reply:
No, surprisingly `perlcritic` doesn't check the syntax. But `perl -c` does and it detects an error on this one. It just happens that `perl -c` is only run on _.pm_ files, no _.pl_ :man_facepalming: 
Can't remember if there is a specific reason for that. I'll look into checking _.pl_ files too

Reply:
See https://github.com/Ensembl/ensembl-compara/commit/37f1e574a1d4e71b6ab41f2c5305ddd5442afeb3